controllers/openstackclient_controller.go

```diff
@@ -546,22 +576,272 @@ func (r *OpenStackClientReconciler) podCreateOrUpdate(instance *ospdirectorv1bet
 
 		err := controllerutil.SetControllerReference(instance, pod, r.Scheme)
 		if err != nil {
+			cond.Message = fmt.Sprintf("Error set controller reference for %s", pod.Name)
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.CommonCondReasonControllerReferenceError)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.CommonCondTypeError)
+			err = common.WrapErrorForObject(cond.Message, instance, err)
+
 			return err
 		}
 
 		return nil
 	})
-	if err != nil && k8s_errors.IsInvalid(err) {
-		// Delete pod when an unsupported change was requested, like
-		// e.g. additional controller VM got up. We just re-create the
-		// openstackclient pod
-		r.Log.Info(fmt.Sprintf("openstackclient pod deleted due to spec change %v", err))
-		if err := r.Client.Delete(context.TODO(), pod); err != nil {
-			return op, err
+	if err != nil {
+		if k8s_errors.IsInvalid(err) {
+			// Delete pod when an unsupported change was requested, like
+			// e.g. additional controller VM got up. We just re-create the
+			// openstackclient pod
+			common.LogForObject(
+				r,
+				fmt.Sprintf("openstackclient pod deleted due to spec change %v", err),
+				instance,
+			)
+			if err := r.Client.Delete(context.TODO(), pod); err != nil {
+
+				return err
```

Comment:
Same question here re: the error.  Any need to log or store it in status?

Reply:
yes we should do that, but ignore k8s_errors.IsNotFound